internal/jptrace/aggregator.go

```diff
@@ -27,7 +35,13 @@ func AggregateTraces(tracesSeq iter.Seq2[[]ptrace.Traces, error]) iter.Seq2[ptra
 				resources := trace.ResourceSpans()
 				traceID := resources.At(0).ScopeSpans().At(0).Spans().At(0).TraceID()
 				if currentTraceID == traceID {
-					mergeTraces(trace, currentTrace)
+					if !skipCurrentTrace {
```

Comment:
there is no need to keep track of skipCurrentTrace, you can have an early exit from mergeTraces if the current count is already at max.